### PR TITLE
Skip gradle check for non code related changes

### PR DESCRIPTION
**Pull request by kotwanikunal**
### Description
- Coming in from #13040 - the file changes are restricted to the changelog file and a GH workflow
 
image

- There is no need to run a `gradle check` in this scenario since the code has not been modified in any way

Pros: Getting to merge in non code related changes without any hassles
Cons(?): Missing out on flakiness

I would love to hear some feedback from other maintainers.

### Check List
- [ ] ~New functionality includes testing.~
  - [ ] ~All tests pass~
- [ ] ~New functionality has been documented.~
  - [ ] ~New functionality has javadoc added~
- [ ] ~Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))~
- [x] Commits are signed per the DCO using --signoff
- [ ] ~Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))~
- [ ] ~Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kotwanikunal** [comment]
Thoughts here - @dblock / @andrross / @reta / @peternied

File: .github/workflows/gradle-check.yml [hunk 1]
```diff
@@ -27,8 +27,27 @@ jobs:
...
+        id: changed-files
+
+      - id: run-gradle-check
+        env:
+          ALL_CHANGED_FILES: ${{ steps.changed-files.outputs.all_changed_files }}
+        run: |
+          run_gradle_check=false
+
+          for filename in ${ALL_CHANGED_FILES}; do
+            # Run gradle check if file ends in [java/gradle/json]
```
Comment (andrross) [inline .github/workflows/gradle-check.yml:40]: I like the spirit here but there are many files that can fail the build that don't meet these extensions:

- no extension, e.g. https://github.com/opensearch-project/OpenSearch/blob/main/libs/core/src/main/resources/META-INF/services/org.opensearch.core.compress.spi.CompressorProvider
- License files (.txt I think?) are checked as a part of the build
- gradle.properties
- probably many many others?

Comment (kotwanikunal) [reply .github/workflows/gradle-check.yml:40]: We can do an inverse set where we skip gradle check if the changes are limited to certain file types

Comment (peternied) [reply .github/workflows/gradle-check.yml:40]: If going this route, I'd rather do a strictly positive list. If the commits only affects: [CHANGELOG*.md, README.md, ...] then skip

Comment (reta) [reply .github/workflows/gradle-check.yml:40]: 👍 to @peternied 's suggestion, other examples are: version.properties, bwcVersions, ....

Comment (kotwanikunal) [reply .github/workflows/gradle-check.yml:40]: That sounds good. I was just bouncing the idea across.
I'll work on refining and testing this out. 
Thanks folks.

**Comment by dblock** [comment]
How about something similar to skip-changelog where a maintainer can make a call that a gradle check is not required?

**Comment by andrross** [comment]
> How about something similar to skip-changelog where a maintainer can make a call that a gradle check is not required?

I mean, we can already do this by just merging the PR without a completed or passing check. Seems a bit like a slippery slope if we normalize manually skipping the gradle check.

**Comment by kotwanikunal** [comment]
> > How about something similar to skip-changelog where a maintainer can make a call that a gradle check is not required?
> 
> I mean, we can already do this by just merging the PR without a completed or passing check. Seems a bit like a slippery slope if we normalize manually skipping the gradle check.

+1 to this. I'd be wary of adding a super power, and instead rely on the framework to guide through it.

File: .github/workflows/gradle-check.yml (hunk 1 above)
Comment (gaiksaya) [reply .github/workflows/gradle-check.yml:40]: I find [paths-ignore](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-excluding-paths) as an easier option. We can keep on adding to the list and also manage file-types, directories, etc. This won't even initiate the workflow run for ignored paths. Example: https://github.com/opensearch-project/OpenSearch/pull/13477

Comment (kotwanikunal) [reply .github/workflows/gradle-check.yml:40]: That sounds good. I'll close out this PR and we can keep adding to `paths-ignore`

**Pull request closed without merging**